components/Programs/Dos.tsx

```diff
@@ -0,0 +1,77 @@
+import styles from '@/styles/Programs/Dos.module.scss';
+
+import type { DosMainFn, DosRuntime } from 'js-dos';
+import type { DosCommandInterface } from 'js-dos/dist/typescript/js-dos-ci';
+import type { AppComponent } from '@/types/utils/programs';
+import type { WindowWithDosModule } from '@/types/components/Programs/dos';
+
+import { useEffect, useRef } from 'react';
+import { getLockedAspectRatioDimensions } from '@/utils/dos';
+import { TITLEBAR_HEIGHT } from '@/utils/constants';
+import { focusClosestFocusableElementFromRef } from '@/utils/elements';
+
+const dosOptions = {
+  wdosboxUrl: '/libs/wdosbox.js',
+  onprogress: () => {} /* eslint @typescript-eslint/no-empty-function: off */
+};
+
+export const loaderOptions = {
+  lockAspectRatio: true,
+  width: 640,
+  height: 400 + TITLEBAR_HEIGHT,
+  bgColor: '#000000'
+};
+
+const Dos: React.FC<AppComponent> = ({
+  args = ['-c', 'CLS'],
+  file: { url, name = '' } = {},
+  maximized
+}) => {
+  let ci: DosCommandInterface;
+  const canvasRef = useRef<HTMLCanvasElement>(null);
+  const loadMain = (main: DosMainFn, prependedArgs: string[] = []) =>
```

Comment:
Move `loadMain` and `loadDos` outside of the component to avoid recreation on re-renders.